src/webui/api/torrentscontroller.cpp

```diff
@@ -873,15 +873,47 @@ void TorrentsController::editTrackerAction()
 
 void TorrentsController::removeTrackersAction()
 {
-    requireParams({u"hash"_s, u"urls"_s});
+    requireParams({u"urls"_s});
 
-    const auto id = BitTorrent::TorrentID::fromString(params()[u"hash"_s]);
-    BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
-    if (!torrent)
-        throw APIError(APIErrorType::NotFound);
+    const QString hashParam = params()[u"hash"_s];
+    const QStringList urlsParam = params()[u"urls"_s].split(u'|', Qt::SkipEmptyParts);
 
-    const QStringList urls = params()[u"urls"_s].split(u'|');
-    torrent->removeTrackers(urls);
+    // remove trackers from specified torrent
+    if (!hashParam.isEmpty())
+    {
+        const auto id = BitTorrent::TorrentID::fromString(hashParam);
+        BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
+        if (!torrent)
+            throw APIError(APIErrorType::NotFound);
+
+        torrent->removeTrackers(urlsParam);
+    }
+    // remove trackers from all torrents
+    else
```

Comment:
For this case, I would suggest using `hash=*` instead of omitting the hash. The intention is clearer with `*` specified.

Reply:
Is there precedent in our web api for using `*` as a sentinel value? I want to be cautious about introducing a new pattern.

Reply:
Is there precedent that omitting a parameter means `all` by default? And for a destructive operation?

Reply:
IMO, if we look back at the existing precedents, we will never improve anything.

Reply:
Was just trying to understand what the current API expected in these cases. Will switch to always requiring a `hash`.